pyintegration/deephaven2/time.py

```diff
@@ -0,0 +1,775 @@
+
+
+def to_datetime(s: str, quiet: bool = False):
+    """ Converts a datetime string to a DateTime object
+
+    Args:
+        s (str): in the form yyyy-MM-ddThh:mm:ss and with optional sub-seconds after an
+            optional decimal point, followed by a mandatory time zone character code
+        quiet (bool, optional): when True, if the datetime string can't be parsed, this function returns None, otherwise
+            it raises an exception; default is False
+
+    Returns:
+        a DateTime
+
+    Raises:
+        DHError
+    """
+    if quiet:
+        return _JDateTimeUtils.convertDateTimeQuiet(s)
+
+    try:
+        return _JDateTimeUtils.convertDateTime(s)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def to_period(s: str, quiet: bool = False) -> Period:
+    """ Converts a period string into a Period object.
+
+    Args:
+        s (str): a string in the form of nYnMnWnDTnHnMnS, with n being numeric values, e.g. 1W for one week, T1M for
+            one minute, 1WT1H for one week plus one hour
+        quiet (bool, optional): when True, if the period string can't be parsed, this function returns None, otherwise
+            it raises an exception; default is False
+
+    Returns:
+        Period
+
+    Raises:
+        DHError
+    """
+    if quiet:
+        return _JDateTimeUtils.convertPeriodQuiet(s)
+
+    try:
+        return _JDateTimeUtils.convertPeriod(s)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def to_nanos(s, quiet: bool = False) -> int:
+    """ Converts a time string to nanoseconds from the Epoch.
+
+    Args:
+        s (str): in the format of: hh:mm:ss[.nnnnnnnnn]
+        quiet (bool): to return None or raise an exception if the string can't be parsed, default is False
+
+    Returns:
+        int
+
+    Raises:
+        DHError
+    """
+    if quiet:
+        return _JDateTimeUtils.convertTimeQuiet(s)
+
+    try:
+        return _JDateTimeUtils.convertTime(s)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def now():
+    """ Provides the current date/time.
+
+    Returns:
+        DateTime
+
+    Raises:
+        DHError
+    """
+    try:
+        return _JDateTimeUtils.currentTime()
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def date_at_midnight(dt, tz: TimeZone):
+    """ Returns a DateTime for the requested DateTime at midnight in the specified time zone.
+
+    Args:
+        dt (DateTime) - DateTime for which the new value at midnight should be calculated
+        tz: (TimeZone) - TimeZone for which the new value at midnight should be calculated
+
+    Returns:
+        DateTime
+
+    Raises:
+        DHError
+    """
+    try:
+
+        return _JDateTimeUtils.dateAtMidnight(dt, tz.value)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def day_of_month(dt, tz: TimeZone) -> int:
+    """ Returns an 1-based int value of the day of the month for a DateTime and specified time zone.
+
+    Args:
```

Comment:
What happens if an input is None?

... same comment on all docs in this module.